Automattic/newspack-plugin
# Configure CircleCI for linting & releasing

### All Submissions:






### Changes proposed in this Pull Request:

This configures:
1. commit message linting, adhering to [Angular's convention](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commit-message-format)
2. CircleCI:
    1. run linting 
    1. release automatically, with [semantic-release](https://semantic-release.gitbook.io/) (from branch `release`, which we will not use until March)

### One more thing

Run `./vendor/brainmaestro/composer-git-hooks/cghooks update` to update git hooks after this is merged 🙏

### How to test the changes in this Pull Request:

1. update git hooks: `./vendor/brainmaestro/composer-git-hooks/cghooks update`
1. add some changes to git staging
2. try to commit with a useless message (e.g. `git commit -m "hehe"`)
3. observe that this shall not pass
4. try to commit with a standardized message (e.g. `git commit -m "feat: increase life span"`)

### Other information:
